x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/model_select.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiSelectable } from '@elastic/eui';
+
+import { MlModel } from '../../../../../../../common/types/ml';
+import { IndexNameLogic } from '../../index_name_logic';
+import { IndexViewLogic } from '../../index_view_logic';
+
+import { MLInferenceLogic } from './ml_inference_logic';
+import { ModelSelectLogic } from './model_select_logic';
+import { ModelSelectOption, ModelSelectOptionProps } from './model_select_option';
+import { normalizeModelName } from './utils';
+
+export const ModelSelect: React.FC = () => {
+  const { indexName } = useValues(IndexNameLogic);
+  const { ingestionMethod } = useValues(IndexViewLogic);
+  const {
+    addInferencePipelineModal: { configuration },
+  } = useValues(MLInferenceLogic);
+  const { selectableModels, isLoading } = useValues(ModelSelectLogic);
+  const { setInferencePipelineConfiguration } = useActions(MLInferenceLogic);
+
+  const { modelID, pipelineName, isPipelineNameUserSupplied } = configuration;
+
+  const getModelSelectOptionProps = (models: MlModel[]): ModelSelectOptionProps[] =>
+    (models ?? []).map((model) => ({
+      ...model,
+      label: model.modelId,
+      checked: model.modelId === modelID ? 'on' : undefined,
+    }));
+
+  const onChange = (options: ModelSelectOptionProps[]) => {
+    const selectedOption = options.find((option) => option.checked === 'on');
+    setInferencePipelineConfiguration({
+      ...configuration,
+      inferenceConfig: undefined,
+      modelID: selectedOption?.modelId ?? '',
+      fieldMappings: undefined,
+      pipelineName: isPipelineNameUserSupplied
+        ? pipelineName
+        : indexName + '-' + normalizeModelName(selectedOption?.modelId ?? ''),
+    });
+  };
+
+  const renderIndexOption = (option: ModelSelectOptionProps) => <ModelSelectOption {...option} />;
+
+  return (
+    <EuiSelectable
+      data-telemetry-id={`entSearchContent-${ingestionMethod}-pipelines-configureInferencePipeline-selectTrainedModel`}
+      options={getModelSelectOptionProps(selectableModels)}
+      singleSelection="always"
+      listProps={{
+        bordered: true,
+        rowHeight: 90,
+        showIcons: false,
+        onFocusBadge: false,
+      }}
+      height={360}
```

Comment:
The static height makes the list look strange when there are fewer than 4 options (see attached screenshot). What if we removed the `height` prop so that the list can resize in response to the number of options?

<img width="1238" alt="Screenshot 2023-12-01 at 11 32 56 AM" src="https://github.com/elastic/kibana/assets/26927948/75ce9bff-833f-4856-b555-17eaa998f46d">

Reply:
I think `min(4, optionsCount) * 90` _almost_ works, but we need to handle two edge cases:

- When rendering for mobile and `rowHeight` > 90
- When filtering models; we need to get the number of results that match the search

In the interest of time, I think we can address both of these in the bug-fix period post-FF. WDYT?